src/utils/appeal/extra.data.ts

```diff
@@ -0,0 +1,43 @@
+import { Attachment } from 'app/models/Attachment';
+import { Illness } from 'app/models/Illness';
+import { OtherReason } from 'app/models/OtherReason';
+import { Reasons } from 'app/models/Reasons';
+import { ReasonType } from 'app/models/fields/ReasonType';
+
+const getReasonType = (reasons: Reasons): ReasonType => {
+    return (ReasonType.illness in reasons && reasons.illness && !reasons.other)
+        ? ReasonType.illness
+        : ReasonType.other;
+};
+
+export const getReasonFromReasons = (reasons: Reasons): Illness | OtherReason | undefined => {
```

Comment:
This is a pretty odd method name.

Reply:
Thanks Joanathan, I am open to any suggestion :)

Reply:
I still don't like this, however improved designs we brainstormed would require refactoring that is outside the scope of this change.